python/ql/src/semmle/python/objects/ObjectAPI.qll

```diff
@@ -412,6 +412,44 @@ class ClassValue extends Value {
         or
         this.hasAttribute("__getitem__")
     }
+    
+    /** Holds if this class is a container(). That is, does it have a __getitem__ method.*/
+    predicate isContainer() {
+        exists(this.lookup("__getitem__"))
+    }
+    
+    /** Holds if this class is probably a sequence. */
+    predicate isSequence() {
+        /* To determine whether something is a sequence or a mapping is not entirely clear,
+         * so we need to guess a bit.
+         */
+        this.getASuperType() = ClassValue::tuple()
+        or
+        this.getASuperType() = ClassValue::list()
+        or
+        this.getASuperType() = ClassValue::rangeType()
+        or
+        this.getASuperType() = ClassValue::bytes()
+        or
+        this.getASuperType() = ClassValue::unicode()
+        or
+        /* Does this inherit from abc.Sequence? */
+        this.getASuperType().getName() = "Sequence"
```

Comment:
```suggestion
        major_version() = 2 and this.getASuperType() = Value::named("collections.Sequence")
        or
        major_version() = 3 and this.getASuperType() = Value::named("collections.abc.Sequence")
```

_note: I did not try to run this code_

References:
- https://docs.python.org/2.7/library/collections.html#collections.Sequence
- https://docs.python.org/3.8/library/collections.abc.html#collections.abc.Sequence

Reply:
Oh, this is a good point (especially if it works!). I agree that the `Value::named` version is preferable (as it mentions the module to which `Sequence` belongs). However, it seems to me that these names are not stable across Python versions (e.g. `collections.Mapping` in Python 2, but `collections.abc.Mapping` in Python 3), so it would probably be best to abstract this out into a separate predicate.

Reply:
... Which is exactly what you did :man_facepalming:. My reading comprehension is not having a good day today.

Reply:
@bekavalentine I think if you accept Rasmus' suggestion, it should be ready to merge. (Assuming you don't disagree with it.)